### PR TITLE
Publish from Travis

### PR DESCRIPTION
**Pull request by 2m**
Adds auto publishing of snapshots and tagged versions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,4 +26,11 @@ before_cache:
 
 notifications:
   email:
-    on_success: never
\ No newline at end of file
+    on_success: never
```
Comment (raboof) [inline .travis.yml]: Are we sure we want this? it's a useful notification for me sometimes :). But this is unrelated to this PR

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+    - stage: test
+      script: sbt test
+    - stage: publish
+      script: sbt publish
+
+stages:
+  - name: test # runs on master commits and PRs
+    if: NOT tag =~ ^v
+  - name: publish # runs on main repo master commits or version-tagged commits
+    if: repo = lightbend/sbt-paradox-apidoc AND ( ( branch = master AND type = push ) OR tag =~ ^v )
```
Comment (raboof) [inline .travis.yml]: So snapshots as well? Do we have a cleanup mechanism for those yet?

Comment (2m) [reply .travis.yml]: No cleanup strategy yet. For smaller projects it probably is not needed, as there are few snapshots.

File: .travis.yml (hunk 1 above)
Comment (2m) [reply .travis.yml]: Yea, I also find these emails useful. I do not disable notifications when creating new `travis.yml` file. But since it was here already, I did not dare to remove it. :)
